### PR TITLE
runc: remove --criu option

### PR DESCRIPTION
**Pull request by kolyshkin**
This was introduced in an initial commit, back in the day when criu was
a highly experimental thing. Today it's not; most users who need it have
it packaged by their distro vendor.

The usual way to run a binary is to look it up in directories listed in
$PATH. This is flexible enough and allows for multiple scenarios (custom
binaries, extra binaries, etc.). This is the way criu should be run.

Make `--criu` a hidden option (thus removing it from help). Remove the
option from man pages, integration tests, etc. Remove all traces of
CriuPath from data structures.

Add a warning that `--criu` is ignored and will be removed.

_This used to be part of #3316._

### CODE REVIEW

**Comment by kolyshkin** [comment]
This should be done 5 years ago but I guess better late than never.

runc also uses `ps`, `busctl`, `systemctl`, `newuidmap`/`newgidmap` etc, but do not provide any options to configure those -- $PATH is sufficient and flexible enough.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Deprecated
```
Comment (AkihiroSuda) [inline CHANGELOG.md:9]: Can be another PR, but maybe we should have `docs/deprecation.md` in addition to [`docs/experimental.md`](https://github.com/opencontainers/runc/blob/master/docs/experimental.md) ?

**Comment by kolyshkin** [comment]
Now, an alternative would be to honor `--criu` but warn that it will be removed. I'd rather not do it, because:
 - most probably no one is using this option;
 - whoever uses it can employ a quick and easy fix:

```diff 
-runc --criu /some/custom/criu/dir/criu ...
+PATH=/some/custom/criu/dir:$PATH runc
```

The only question is, should using `--criu` result in an error  (rather than a warning).

File: CHANGELOG.md (hunk 1 above)
Comment (kolyshkin) [reply CHANGELOG.md:9]: I don't know. It seems easy enough to search CHANGELOG.md for `eprecat`.

**Comment by rhatdan** [comment]
LGTM

**Comment by thaJeztah** [comment]
I guess this is fine, as the flag is still present (so things won't hard-fail).

I stumbled upon this option in containerd, which (I think) may be using this flag; https://github.com/containerd/containerd/blob/77d53d2d230c3bcd3f02e6f493019a72905c875b/cmd/containerd-shim/main_unix.go#L60

Originally added in https://github.com/containerd/containerd/commit/ab0cb4e756206158e32e1a5c7e6c00b6a73b30e8, and I can see the `--criu` flag being set in containerd/go-runc; https://github.com/containerd/go-runc/blob/1c99ade6c04ed41bc4c9a8ac12b81b5fd4e14bcc/runc.go#L730-L732

So that code probably should be removed, and the option deprecated on containerd's side as well, @AkihiroSuda

**Comment by thaJeztah** [comment]
Opened https://github.com/containerd/go-runc/pull/82 and https://github.com/containerd/containerd/pull/6496

**Review by crosbymichael — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@AkihiroSuda @cyphar good to go?

**Comment by kolyshkin** [comment]
@adrianreber PTAL

**Review by AkihiroSuda — APPROVED** [review]
